manifests/slave.pp

```diff
@@ -174,13 +174,38 @@
       $defaults_group = 'root'
       $manage_user_home = true
 
-      file { '/etc/init.d/jenkins-slave':
-        ensure => 'file',
-        mode   => '0755',
-        owner  => 'root',
-        group  => 'root',
-        source => "puppet:///modules/${module_name}/jenkins-slave.${::osfamily}",
-        notify => Service['jenkins-slave'],
+      $systemd_min_release = $::osfamily ? {
+        'RedHat' => $::operatingsystem ? {
+          'Fedora' => 21,
+          default  => 7,
+        },
+        default => undef,
+      }
+      if $systemd_min_release {
+        $have_systemd = $::operatingsystemmajrelease >= $systemd_min_release
```

Comment:
I would be OK with adding a dep on that module.  I know if it would or wouldn't be a better approach, just pointing it out as an implementation option.